src/sources/eventstoredb_metrics/types.rs

```diff
@@ -0,0 +1,232 @@
+use crate::event::{Metric, MetricKind, MetricValue};
+use serde::de::{MapAccess, Visitor};
+use serde::{Deserialize, Deserializer};
+use std::collections::BTreeMap;
+
+#[derive(Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct Stats {
+    pub proc: Proc,
+    pub sys: Sys,
+}
+
+impl Stats {
+    pub fn metrics(&self, namespace: Option<String>) -> Vec<Metric> {
+        let mut result = Vec::new();
+        let mut tags = BTreeMap::new();
+        let now = chrono::Utc::now();
+        let namespace = namespace.unwrap_or_else(|| "eventstoredb".to_string());
+
+        tags.insert("id".to_string(), self.proc.id.to_string());
+
+        result.push(
+            Metric::new(
+                "memory_usage",
+                MetricKind::Absolute,
+                MetricValue::Gauge {
+                    value: self.proc.mem as f64,
+                },
+            )
+            .with_namespace(Some(namespace.clone()))
+            .with_tags(Some(tags.clone()))
+            .with_timestamp(Some(now)),
+        );
+
+        result.push(
+            Metric::new(
+                "disk_io_read_bytes",
```

Comment:
We have been trying to follow the [prometheus naming conventions](https://prometheus.io/docs/practices/naming/) for metrics. That would make this `disk_io_read_bytes_total` since it is a counter. Would you mind updating this and the other metric names?